plutus-ledger-api/src/Plutus/ApiCommon.hs

```diff
@@ -343,3 +347,52 @@ evaluateScriptCounting lv pv verbose ectx p args = swap $ runWriter @LogOutput $
     tell logs
     liftEither $ first CekError $ void res
     pure final
+
+
+{-| A valid parameter name has to be enumeration, bounded, ordered, and
+prettyprintable in a "lowerKebab" way.
+
+Each API version should expose such an enumeration as an ADT and create
+an instance of ParamName out of it.
+-}
+class (Enum a, Ord a, Bounded a) => IsParamName a where
+   showParamName :: a -> String
+   default showParamName :: (Generic a, GIsParamName (Rep a)) => a -> String
+   showParamName = gshowParamName . from
+
+-- | A datatype-generic class to prettyprint 'sums of nullary constructors' in lower-kebab syntax.
+class GIsParamName f where
+    gshowParamName :: f p -> String
+
+instance (GIsParamName a) => GIsParamName (M1 D i a) where
+    gshowParamName (M1 x) = gshowParamName x
+
+instance Constructor i => GIsParamName (M1 C i U1) where
+    gshowParamName = lowerKebab . conName
+      where
+        lowerKebab :: String -> String
+        lowerKebab (h:t) = toLower h : fmap (\c -> if c == '\''
```

Comment:
how about using `_` instead of `'`? `_` is valid in identifiers and it looks a bit more "like" the output, i.e. `Foo_Bar` -> `Foo-Bar`

Reply:
also you're violating my favourite rule again :)

Reply:
I had indeed `_` but guess what, we renamed sha2 and blake to sha2_256 and blake2_256. Guess how it horribly breaks

Reply:
I was actually desperate, until I dug up the haskell98 and got relieved :  `'` is valid in the bnf

Reply:
Oh lol of course

Reply:
Worth a note